taca/illumina/__init__.py

```diff
@@ -0,0 +1,172 @@
+                and barcode_missmatches in range(3):
+            cl.extend(['--barcode-missmatches', barcode_missmatches])
+        if cl_options.get('create-fastq-for-index-reads'):
+            cl.append('--create-fastq-for-index-reads')
+        if cl_options.get('ignore-missing-bcls'):
+            cl.append('--ignore-missing-bcls')
+        if cl_options.get('ignore-missing-filter'):
+            cl.append('--ignore-missing-filter')
+        if cl_options.get('ignore-missing-locs'):
+            cl.append('--ignore-missing-locs')
+        mask = cl_options.get('mask-short-adapter-reads')
+        if mask and type(mask) is int:
+            cl.extend(['--mask-short-adapter-reads', mask])
+        minimum = cl_options.get('minimum-trimmed-reads')
+        if minimum and type(minimum) is int:
+            cl.extend(['--minimum-trimmed-reads', minimum])
+        if cl_options.get('tiles'):
+            cl.extend(['--tiles', cl_options.get('tiles')])
+
+        if cl_options.get('use-bases-mask'):
+            cl.extend(['--use-bases-mask', cl_options.get('use-bases-mask')])
+
+        if cl_options.get('with-failed-reads'):
+            cl.append('--with-failed-reads')
+        if cl_options.get('write-fastq-reverse-complement'):
+            cl.append('--write-fastq-reverse-complement')
+
+        LOG.info(("BCL to FASTQ conversion and demultiplexing started for "
+                  " run {} on {}".format(os.path.basename(run), datetime.now())))
+
+        misc.call_external_command_detached(cl, with_log_files=True)
+
+        LOG.info(("BCL to FASTQ conversion and demultiplexing finished for "
+                  "run {} on {}".format(os.path.basename(run), datetime.now())))
+
+
+def demultiplex_HiSeq(run):
+    """ Demultiplexing for HiSeq (V3/V4) runs
+    """
+    raise NotImplementedError('Meec! Demultiplexing for HiSeq (V3/V4) runs not implemented yet :-/')
+
+
+def demultiplex_MiSeq(run):
+    """ Demultiplexing for MiSeq runs
+    """
+    raise NotImplementedError('Meec! Demultiplexing for MiSeq runs not implemented yet :-/')
+
+
+class Run(object):
+    """ Defines an Illumina run
+    """
+    def __init__(self, run_dir):
+        if not os.path.exists(run_dir) or not \
+                os.path.exists(os.path.join(run_dir, 'runParameters.xml')):
+            raise RuntimeError('Could not locate run directory {}'.format(run_dir))
+        self.run_dir = run_dir
+        self.id = os.path.basename(run_dir)
+        self._extract_run_info()
+
+        if self.run_type == 'HiSeq':
+            self.demultiplex = demultiplex_HiSeq
+        elif self.run_type == 'HiSeq X':
+            self.demultiplex = demultiplex_HiSeq_X
+        elif self.demultiplex == 'MiSeq':
+            self.demultiplex = demultiplex_MiSeq
+
+
+    def _extract_run_info(self):
+        """ Extracts run info from runParameters.xml and adds it to the class attributes
+
+        TODO: This method could be used to extract A LOT of information about the
```

Comment:
@guillermo-carrasco this would be great, and would allow to populate status db from preprocessing server, and avoid all the confusion that is now on a2010002. 
We need to decide if these stats will go to an Xten FC database or to the flowcell database... I am more inclined for the first one.